### PR TITLE
fix: Initialise clients only once

### PR DESCRIPTION
**Pull request by yevgenypats**
This fix a panic that was caused if plugins misbehave and return difference results for difference runs with the same config.

### CODE REVIEW

**Comment by yevgenypats** [comment]
Still testing this

**Review by erezrokah — CHANGES_REQUESTED** [review]
The fix looks good, but I won't kill the process because we can't send metrics. Maybe only log it?

**Comment by yevgenypats** [comment]
I think there were number of issues here and after fixing the initial issue with making sure `Multiplex` is called only once to combat with clients that return random clients on each call I was hit again by the flattening and global list. I'll look into refactoring this somehow next week

File: plugins/source.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ type SourcePlugin struct {
 	// Tables is all tables supported by this source plugin
 	tables schema.Tables
 	// status sync metrics
-	metrics SourceMetrics
+	metrics *SourceMetrics
```
Comment (erezrokah) [inline plugins/source.go:29]: Is this change to a pointer required for the fix?

Comment (yevgenypats) [reply plugins/source.go:29]: I think yes

**Review by erezrokah — APPROVED** [review]

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**